core/templates/pages/exploration-editor-page/modal-templates/state-diff-modal.component.ts

```diff
@@ -0,0 +1,106 @@
+ */
+
+import { OnInit } from '@angular/core';
+import { Component } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { ConfirmOrCancelModal } from 'components/common-layout-directives/common-elements/confirm-or-cancel-modal.component';
+import { State } from 'domain/state/StateObjectFactory';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { ContextService } from 'services/context.service';
+import { StateDiffModalBackendApiService } from '../services/state-diff-modal-backend-api.service';
+
+@Component({
+  selector: 'state-diff-modal',
+  templateUrl: './state-diff-modal.component.html',
+})
+export class StateDiffModalComponent
+  extends ConfirmOrCancelModal implements OnInit {
+    newState: State | null;
+    oldState: State | null;
+    newStateName: string;
+    oldStateName: string;
+    headers: {
+        leftPane: string;
+        rightPane: string;
+    };
+    yamlStrs: {
+        leftPane: string;
+        rightPane: string;
+    } = {
+      leftPane: '',
+      rightPane: '',
+    };
+    CODEMIRROR_MERGEVIEW_OPTIONS = {
+      lineNumbers: true,
+      readOnly: true,
+      mode: 'yaml',
+      viewportMargin: 100
+    };
+
+    constructor(
+        private ngbActiveModal: NgbActiveModal,
+        private contextService: ContextService,
+        private urlInterpolationService: UrlInterpolationService,
+        private stateDiffModalBackendApiService:
+        StateDiffModalBackendApiService,
+    ) {
+      super(ngbActiveModal);
+    }
+
+    ngOnInit(): void {
+      const url = this.urlInterpolationService.interpolateUrl(
+        '/createhandler/state_yaml/<exploration_id>', {
+          exploration_id: (
+            this.contextService.getExplorationId())
+        });
+
+      if (this.oldState) {
+        this.stateDiffModalBackendApiService.fetchYaml(
+          this.oldState.toBackendDict(), 50, url).then(response => {
+          this.yamlStrs.leftPane = response.data.yaml;
+        });
+      } else {
+        // Note: the timeout is needed or the string will be sent
+        // before codemirror has fully loaded and will not be
+        // displayed. This causes issues with the e2e tests.
+        setTimeout(() => {
+          this.yamlStrs.leftPane = '';
+        }, 200);
+      }
+
+      if (this.newState) {
+        this.stateDiffModalBackendApiService.fetchYaml(
+          this.newState.toBackendDict(), 50, url).then(response => {
+          this.yamlStrs.rightPane = response.data.yaml;
+        });
+      } else {
+        // Note: the timeout is needed or the string will be sent
+        // before codemirror has fully loaded and will not be
+        // displayed. This causes issues with the e2e tests.
+        setTimeout(() => {
+          this.yamlStrs.rightPane = '';
+        }, 200);
+      }
+    }
+}
+
+angular.module('oppia').controller(
```

Comment:
You don't need to downgrade this component.
Have you tested your changes locally?

Reply:
It is being invoked in version-diff-visualization.component.ts in the function ctrl.showStateDiffModal() using $uibModal.open(). Do I have to change that?
I tested the changes locally before making the PR and the exploration-editor-page was working fine. I will check again.